File: easy/0495_teemo_attacking/solution.py
from typing import List


class Solution:
    def findPoisonedDuration(
        self, time_series: List[int], duration: int
    ) -> int:
        time_series.append(time_series[-1] + duration)

        poisoned_duration = 0

        for i in range(1, len(time_series)):
            poisoned_duration += min(
                duration, time_series[i] - time_series[i - 1]
            )

        return poisoned_duration

File: easy/0495_teemo_attacking/solution.cpp
#include <vector>

class Solution {
public:
  int findPoisonedDuration(std::vector<int> &timeSeries, int duration) {
    timeSeries.push_back(timeSeries.back() + duration);

    int poisonedDuration = 0;

    for (size_t i = 1; i < timeSeries.size(); ++i)
      poisonedDuration += std::min(duration, timeSeries[i] - timeSeries[i - 1]);

    return poisonedDuration;
  }
};
